File: 344. Reverse String/solution.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function (s) {
  let first = 0;
  let last = s.length - 1;

  while (first < last) {
    let temp = s[first];
    s[first] = s[last];
    s[last] = temp;
    first++;
    last--;
  }
};

arr = ["h", "e", "l", "l", "o"];
console.log(arr);
reverseString(arr);
console.log(arr);

File: 344. Reverse String/Solution.java
public class Solution {
    public void reverseString(String[] s) {
        int first = 0;
        int last = s.length - 1;
        while (first < last) {
            String temp = s[first];
            s[first] = s[last];
            s[last] = temp;
            first++;
            last--;
        }

    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String[] s = { "h", "e", "l", "l", "o" };
        solution.reverseString(s);

        for (String string : s) {
            System.out.print(string + " ");
        }
    }
}
